### PR TITLE
Handling a corner case in discovery.

### PR DESCRIPTION
**Pull request by alainjobart**
If only two tablets are there, with replication lag of 3h and 30h, we'd
return nothing. We want to return both.

### CODE REVIEW

**Comment by alainjobart** [comment]
@mpawliszyn this is a follow-up to your PR https://github.com/youtube/vitess/pull/3383 as our internal tests uncovered a corner case.

File: go/vt/discovery/replicationlag.go [hunk 1]
```diff
@@ -108,6 +108,20 @@ func filterByLag(tabletStatsList []*TabletStats) []*TabletStats {
...
+		// different enough that the 70% mean computation up
+		// there didn't find them all in a group. For
+		// instance, if *minNumTablets = 2, and we have two
+		// tablets with lag of 3h and 30h.
+		// In that case, we just use them all.
+		for _, ts := range list {
+			snapshots = append(snapshots, tabletLagSnapshot{
+				ts:     ts,
+				replag: ts.Stats.SecondsBehindMaster})
+		}
```
Comment (sougou) [inline go/vt/discovery/replicationlag.go]: I thought we should never return a tablet that's lagged by greater than the 'very high' threshold, because the data is too stale. This seems to contradict that.

**Comment by alainjobart** [comment]
Looking more at this, I'm uncovering more troublesome code. In this area:
https://github.com/youtube/vitess/blob/master/go/vt/vttablet/tabletmanager/healthcheck.go#L213
it seems we have two other flags on the vttablet side, that seem to do the same thing:

* the degraded mode flag is only used for display.
* the unhealthy flag is used to report unhealthy if replication lag is higher than 2h.
* if replication is stopped, we report healthy, with 2h lag exactly.

So vtgate will never see tablets with more than 2h lag. Given that, I suggest the following course of action:

* remove the 2h flag from vtgate entirely. Let the vttablet side control that, no need to have it in two places.
* keep the vttablet behavior as is, especially the replication stopped -> 2h delay. This would address the corner case of a master failure, no slave is replicating, we still serve.
* adjust the tests so we only test valid cases, with reported values < 2h.

What do yo think?

**Comment by sougou** [comment]
LGTM

**Comment by alainjobart** [comment]
@sougou reworked the test a bit, and added a comment. PTAL
